consensus/spos/bls/subroundEndRound.go

```diff
@@ -302,39 +302,28 @@ func (sr *subroundEndRound) doEndRoundJobByLeader() bool {
 	sig, err := sr.SignatureHandler().AggregateSigs(bitmap, sr.Header.GetEpoch())
```

Comment:
I think you can extract L301-L328 in a separate method which will return bitmap, sig and err, actually you can directly return from there sr.handleInvalidSignersOnAggSigFail(), and make the code easier to read there without so much if else levels. Here you should have only something like this:

```
bitmap, sig, err := sr.aggregateSigsAndHandleInvalidSigners(bitmap)
if err != nil {
	log.Debug("doEndRoundJobByLeader.aggregateSigsAndHandleInvalidSigners", "error", err.Error())
	return false
}

```